core/controller/src/main/scala/org/apache/openwhisk/core/loadBalancer/ShardingContainerPoolBalancer.scala

```diff
@@ -179,14 +179,53 @@ class ShardingContainerPoolBalancer(
     TrieMap[ActivationId, Promise[Either[ActivationId, WhiskActivation]]]()
   private val activationsPerNamespace = TrieMap[UUID, LongAdder]()
   private val totalActivations = new LongAdder()
-  private val totalActivationMemory = new LongAdder()
+  private val totalBBActivationMemory = new LongAdder()
+  private val totalManagedActivationMemory = new LongAdder()
 
   /** State needed for scheduling. */
   protected[loadBalancer] val schedulingState = ShardingContainerPoolBalancerState()(lbConfig)
 
   actorSystem.scheduler.schedule(0.seconds, 10.seconds) {
     MetricEmitter.emitHistogramMetric(LOADBALANCER_ACTIVATIONS_INFLIGHT(controllerInstance), totalActivations.longValue)
-    MetricEmitter.emitHistogramMetric(LOADBALANCER_MEMORY_INFLIGHT(controllerInstance), totalActivationMemory.longValue)
+    MetricEmitter.emitHistogramMetric(
+      LOADBALANCER_MEMORY_INFLIGHT(controllerInstance, ""),
+      totalBBActivationMemory.longValue + totalManagedActivationMemory.longValue)
+    MetricEmitter.emitHistogramMetric(
+      LOADBALANCER_MEMORY_INFLIGHT(controllerInstance, "Blackbox"),
+      totalBBActivationMemory.longValue)
+    MetricEmitter.emitHistogramMetric(
+      LOADBALANCER_MEMORY_INFLIGHT(controllerInstance, "Managed"),
+      totalManagedActivationMemory.longValue)
+    MetricEmitter.emitHistogramMetric(
+      INVOKER_TOTALMEM_BLACKBOX,
+      schedulingState.blackboxInvokers.filter(_.status.isUsable).map(_.id.userMemory.toMB).sum)
```

Comment:
this needs to traverse the whole invoker list 3 times (filter, map, sum). As a general rule, filter + map can always be concatenated into one `collect`, in this case:

```scala
schedulingState.blackboxInvokers.collect { 
  case in if in.status.isUsable => in.id.userMemory.toMB 
}.sum
```

However, in this specific case, we should be able to reduce the traversals to one time.

```scala
schedulingState.blackboxInvokers.foldLeft(0L) { (acc, curr) => if(curr.status.isUsable) {
    acc + curr.id.userMemory.toMB
  } else {
    acc
  }
}
```

(Pseudo code, no IDE installed locally)